app/scripts/metamask-controller.js

```diff
@@ -772,9 +772,15 @@ export default class MetamaskController extends EventEmitter {
       await this.diagnostics.reportMultipleKeyrings(nonSimpleKeyrings)
     }
 
-    await this.blockTracker.checkForLatestBlock()
+    try {
+      await this.blockTracker.checkForLatestBlock()
+    } catch (error) {
+      log.error('Error while unlocking extension.', error)
+    }
 
     try {
+      await this.blockTracker.checkForLatestBlock()
```

Comment:
Why is this now being called twice? :thinking:

Reply:
Because I'm a dumb-dumb. Removed duplicate call here: ae46338

I make the call in a separate try/catch block because I don't the latest block and 3box workflows to interfere with each other.